healthtip/web.go

```diff
@@ -47,6 +53,45 @@ func Run(dbCon *sql.DB) error {
 	return nil
 }
 
+func authMiddleware(next func(w http.ResponseWriter, r *http.Request)) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		sharedKeyPath := os.Getenv("AUTH0_PUBLIC_KEY_PATH")
+		audience := []string{os.Getenv("AUTH0_AUDIENCE")}
+		sharedKey, err := ioutil.ReadFile(sharedKeyPath)
+		if err != nil {
+			log.Println("PEM file at " + sharedKeyPath + " could not be read")
+			handleError(w, r, http.StatusInternalServerError, "Internal error")
+			return
+		}
+		secret, err := joseutil.LoadPublicKey(sharedKey)
+		if err != nil {
+			log.Println("Could not load public key from PEM file.")
+			handleError(w, r, http.StatusInternalServerError, "Internal error")
+			return
+		}
+		secretProvider := auth0.NewKeyProvider(secret)
+		configuration := auth0.NewConfiguration(secretProvider, audience, "https://"+os.Getenv("AUTH0_DOMAIN")+".auth0.com/", jose.RS256)
+		validator := auth0.NewValidator(configuration, nil)
+
+		token, err := validator.ValidateRequest(r)
```

Comment:
What would happen if this is invalid? Should we throw 401 Unauthorized if there's a validation error, or similarly if the claims.Subject split doesn't have enough split parts?

Reply:
Ah I guess I see the error checking below, it should probably be moved up.

Reply:
Sure I can make this change.